src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -367,6 +363,13 @@ public virtual string CustomMethod
         [Parameter]
         public virtual SwitchParameter PreserveHttpMethodOnRedirect { get; set; }
 
+        /// <summary>
+        /// Gets or sets the UnixSocket property.
+        /// </summary>
+        [Parameter]
+        [ValidateNotNullOrEmpty]
+        public virtual UnixDomainSocketEndPoint UnixSocket { get; set; }
```

Comment:
Should this be a runtime check for Windows or `#ifdef` so it doesn't exist on Windows at all?

Reply:
https://devblogs.microsoft.com/commandline/windowswsl-interop-with-af_unix/
https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/

From these articles it seems that windows 10 supports UnixSockets, did I misunderstand?

Reply:
Ok, wasn't aware of this!  Very cool.  Have you tested this manually?

Reply:
It works correctly in WSL1, I wasn't able to test it in Windows. I'll try to look into it

Reply:
@SteveL-MSFT It works correctly in Windows 10 (maybe we should throw an error on windows 7-8).

server code:
```cs
System.IO.Directory.CreateDirectory("c://tmp");
const string UnixSocketPath = "c://tmp//foo.sock";

if (File.Exists(UnixSocketPath))
{
    File.Delete(UnixSocketPath);
}

var builder = WebApplication.CreateBuilder(args);
builder.WebHost.ConfigureKestrel(options =>
{
    options.ListenUnixSocket(UnixSocketPath);
});

var app = builder.Build();
app.MapGet("/", () => "Hello from Foo API served via Unix Socket.");

app.Run();
```

inspired by:
https://medium.com/codex/unix-domain-sockets-in-net-6-basics-and-real-world-examples-8982898ab293

client code:
`Invoke-WebRequest -Uri "http://127.0.0.0/" -UnixSocket "/tmp/foo.sock"`
`curl --unix-socket /tmp/foo.sock 127.0.0.0`

Reply:
We still need to support Windows Server 2012/R2, is this API supported on this platform?

Reply:
We should throw an exception when `Socket.OSSupportsUnixDomainSockets is false`